pkg/trace/traceutil/azure.go

```diff
@@ -71,27 +71,46 @@ func getAppServicesTags(getenv func(string) string) map[string]string {
 		aasSiteType:        appService,
 		aasSubscriptionID:  subscriptionID,
 	}
+
+	// Remove the Java and .NET logic once non-universal extensions are deprecated
+	if websiteOS == "windows" {
+		if len(extensionVersion) > 0 {
+			tags[aasExtensionVersion] = extensionVersion
+		} else if hasEnv("DD_AAS_JAVA_EXTENSION_VERSION", getenv) {
+			tags[aasExtensionVersion] = getenv("DD_AAS_JAVA_EXTENSION_VERSION")
+		} else if hasEnv("DD_AAS_DOTNET_EXTENSION_VERSION", getenv) {
+			tags[aasExtensionVersion] = getenv("DD_AAS_DOTNET_EXTENSION_VERSION")
+		}
+	}
+
+	return tags
 }
 
 func getEnvOrUnknown(env string, getenv func(string) string) string {
 	val := getenv(env)
-	if len(env) == 0 {
+	if len(val) == 0 {
 		val = unknown
 	}
 	return val
 }
 
 func getRuntime(getenv func(string) string) (rt string) {
-	env := getenv(ddRuntime)
-	switch env {
-	case dotnetRuntime:
-		rt = dotnetFramework
-	case nodeRuntime:
+	rt = unknown
+
+	env := getenv("WEBSITE_STACK")
+	if env == "JAVA" {
+		rt = javaFramework
+	} else if env == "NODE" || hasEnv("WEBSITE_NODE_DEFAULT_VERSION", getenv) {
 		rt = nodeFramework
-	default:
-		rt = unknown
+	} else if hasEnv("DOTNET_CLI_TELEMETRY_PROFILE", getenv) {
```

Comment:
Are we sure this works for both ".NET Framework" and ".NET Core" apps?

Reply:
I checked some other apps and looks like `DOTNET_CLI_TELEMETRY_PROFILE ` and `DOTNET_SKIP_FIRST_TIME_EXPERIENCE` are present in apps with "Dotnet - vX.0" and "Dotnetcore" runtimes

Reply:
Thank for confirming! Looks like AAS applies them to both even though `DOTNET_SKIP_FIRST_TIME_EXPERIENCE` only has any effect on .NET Core. Resolving this thread.